src/Microsoft.Health.Fhir.Shared.Api/Controllers/ValidateController.cs

```diff
@@ -0,0 +1,70 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System.Linq;
+using System.Net;
+using System.Threading.Tasks;
+using Hl7.Fhir.Model;
+using MediatR;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Options;
+using Microsoft.Health.Fhir.Api.Features.ActionResults;
+using Microsoft.Health.Fhir.Api.Features.Audit;
+using Microsoft.Health.Fhir.Api.Features.Filters;
+using Microsoft.Health.Fhir.Api.Features.Routing;
+using Microsoft.Health.Fhir.Api.Features.Security;
+using Microsoft.Health.Fhir.Core.Configs;
+using Microsoft.Health.Fhir.Core.Extensions;
+using Microsoft.Health.Fhir.Core.Messages.Operation;
+using Microsoft.Health.Fhir.ValueSets;
+
+namespace Microsoft.Health.Fhir.Api.Controllers
+{
+    [ServiceFilter(typeof(AuditLoggingFilterAttribute))]
+    [ServiceFilter(typeof(OperationOutcomeExceptionFilterAttribute))]
+    [ServiceFilter(typeof(ValidateContentTypeFilterAttribute))]
+    [ValidateResourceTypeFilter]
+    [ValidateModelState]
+    [Authorize(PolicyNames.FhirPolicy)]
+    public class ValidateController : Controller
+    {
+        private readonly IMediator _mediator;
+        private readonly CoreFeatureConfiguration _coreFeatures;
+
+        public ValidateController(IMediator mediator, IOptions<CoreFeatureConfiguration> coreFeatures)
+        {
+            _mediator = mediator;
+            _coreFeatures = coreFeatures.Value;
+        }
+
+        [HttpPost]
+        [Route(KnownRoutes.ValidateResourceType)]
+        [AuditEventType(AuditEventSubType.Read)]
+        [Authorize(PolicyNames.ReadPolicy)]
+        public async Task<IActionResult> Validate([FromBody] Resource resource)
+        {
+            if (!_coreFeatures.SupportsValidate || resource.ResourceType == ResourceType.Parameters)
```

Comment:
There also seem to be some parameters for `$validate` that we don't support, so we may want to also let the user know of those - https://www.hl7.org/fhir/resource-operation-validate.html

Reply:
Added exception responses if the user passes a mode or profile.